### PR TITLE
support multiple dirs (grep)

### LINKED ISSUE #440: Specify list of directories for live_grep.
**Is your feature request related to a problem? Please describe.**
Want to do a grep on list of folders only not all folders.
This works for find_files by either
find_files find_command=rg,--files,[list of dir]
or looks like added as a feature in https://github.com/nvim-telescope/telescope.nvim/issues/238

Would love if similar thing is available for live_grep.

### PR DESCRIPTION
**Pull request by tami5**
closes #440

adds search_dirs and vimgrep_arguments opts.

### CODE REVIEW

File: lua/telescope/builtin/files.lua [hunk 1]
```diff
@@ -53,19 +62,29 @@ end
...
 --  opts.search -- the string to search.
+--  opts.search_dirs -- list of directory to search in
 files.grep_string = function(opts)
   -- TODO: This should probably check your visual selection as well, if you've got one
-  local search = opts.search or vim.fn.expand("<cword>")
-
-  search = escape_chars(search)
-
+  opts.vimgrep_arguments = opts.vimgrep_arguments or conf.vimgrep_arguments
+  opts.search = opts.search or vim.fn.expand("<cword>")
```
Comment (Conni2461) [inline lua/telescope/builtin/files.lua:69]: Lets make both not part of opts. No real reason for that. Lets keep local. Same for `live_grep`

File: lua/telescope/builtin/files.lua [hunk 2]
```diff
@@ -22,11 +21,21 @@ local escape_chars = function(string)
...
 files.live_grep = function(opts)
+  opts.vimgrep_arguments = opts.vimgrep_arguments or conf.vimgrep_arguments
+
   if opts.cwd then
     opts.cwd = vim.fn.expand(opts.cwd)
   end
 
+  if opts.search_dirs then
+    for i, path in ipairs(opts.search_dirs) do
+      opts.search_dirs[i] = vim.fn.expand(path)
```
Comment (Conni2461) [inline lua/telescope/builtin/files.lua:35]: Also lets not change the dict that is passed in make a new local for that

File: lua/telescope/builtin/files.lua [hunk 3]
```diff
@@ -53,19 +62,29 @@ end
...
+    end
+  end
 
   pickers.new(opts, {
     prompt_title = 'Find Word',
     finder = finders.new_oneshot_job(
-      flatten { conf.vimgrep_arguments, opts.word_match, search, '.' },
+      flatten {
+        opts.vimgrep_arguments,
+        opts.word_match or nil,
```
Comment (Conni2461) [inline lua/telescope/builtin/files.lua:84]: `or nil` is not necessary here

File: lua/telescope/builtin/files.lua (hunk 1 above)
Comment (tami5) [reply lua/telescope/builtin/files.lua]: done

File: lua/telescope/builtin/files.lua (hunk 2 above)
Comment (tami5) [reply lua/telescope/builtin/files.lua]: done

File: lua/telescope/builtin/files.lua (hunk 3 above)
Comment (tami5) [reply lua/telescope/builtin/files.lua]: done

**Review by Conni2461 — APPROVED** [review]
I wanted to keep `search` in `grep_string` a local as well but whatever. 
LGTM feel free to merge

**Comment by tami5** [comment]
> I wanted to keep `search` in `grep_string` a local as well but whatever.
> LGTM feel free to merge

Can you take another look?

**Review by Conni2461 — APPROVED** [review]
LGTM. Merge if you are done

**Pull request merged**